### PR TITLE
feat: Add guild onboarding settings

### PR DESCRIPTION
**Pull request by jollyroger182**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Add `fetch_onboarding` method to `Guild` objects to allow viewing and modifying guild onboarding settings. Usage example:

```python
onboarding = await guild.fetch_onboarding()
await onboarding.edit(prompts=[], default_channel_ids=[], enabled=False, mode=0)
assert not (onboarding.prompts or onboarding.default_channel_ids or onboarding.enabled)
emoji = PartialEmoji(name=CUSTOM_EMOJI_NAME, id=CUSTOM_EMOJI_ID)
options = [
    OnboardingPromptOption.create('Option 1', role_ids=[ROLE_ID_1], emoji=emoji),
    OnboardingPromptOption.create('Option 2', role_ids=[ROLE_ID_2], emoji='⭐'),
]
prompt = OnboardingPrompt.create(title='Question', options=options, required=True, single_select=True)
await onboarding.edit(prompts=[prompt], default_channel_ids=[CATEGORY_ID], enabled=True, mode=1)
assert (
    onboarding.enabled
    and len(onboarding.prompts) == 1
    and len(onboarding.prompts[0].options) == 2
    and onboarding.prompts[0].title == 'Question'
    and onboarding.prompts[0].required
    and onboarding.prompts[0].single_select
    and onboarding.prompts[0].options[0].title == 'Option 1'
)
await onboarding.edit(enabled=False)
```

## Changes

- Add `get_guild_onboarding` and `modify_guild_onboarding` methods to `GuildRequests`
- Add `Onboarding`, `OnboardingPrompt`, and `OnboardingPromptOption` classes
- Add `OnboardingMode` and `OnboardingPromptType` enums
- Add `fetch_onboarding` method to `Guild`

## Related Issues

#1645 

## Test Scenarios

Try to edit the onboarding settings for a guild and ensure they are changed (similar to the example given above)

## Python Compatibility

- [x] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [x] I've added tests for my code, if applicable
- [x] I've updated / added  documentation, where applicable

### CODE REVIEW

**Review by eightween — APPROVED** [review]

**Comment by jollyroger182** [comment]
I just rebased from the current interactions-py:unstable branch to keep this PR up-to-date.

**Comment by eightween** [comment]
Waiting for more approvals before a merge can be done

**Review by silasary — APPROVED** [review]

**Pull request merged**
